src/Akka.Streams.Kafka/Stages/Producers/DefaultProducerStage.cs

```diff
@@ -233,7 +232,18 @@ private void HandleProduceError(IProducer<K, V> producer, Error error)
             if (!KafkaExtensions.IsBrokerErrorRetriable(error) && !KafkaExtensions.IsLocalErrorRetriable(error))
             {
                 var exception = new KafkaException(error);
-                FailStage(exception);
+                switch (_decider(exception))
+                {
+                    case Directive.Stop:
+                        if (_stage.CloseProducerOnStop)
+                        {
+                            producer.Dispose();
+                        }
+                        _failStageCallback(exception);
```

Comment:
Need to have `Directive.Restart` support too - but it is better then what we had anyway.